examples/howto/optimize_evoked.py

```diff
@@ -42,82 +43,183 @@
+                          numspikes=1,
+                          location='proximal',
+                          weights_ampa=weights_ampa_p1,
+                          weights_nmda=weights_nmda_p1,
+                          synaptic_delays=synaptic_delays_p)
+
+# Distal
+weights_ampa_d1 = {'L2_basket': 0.8037, 'L2_pyramidal': 0.5738,
+                   'L5_pyramidal': 0.3626}
+weights_nmda_d1 = {'L2_basket': 0.2492, 'L2_pyramidal': 0.6183,
+                   'L5_pyramidal': 0.1121}
+synaptic_delays_d1 = {'L2_basket': 0.1, 'L2_pyramidal': 0.1,
+                      'L5_pyramidal': 0.1}
+net_init.add_evoked_drive('evdist1',
+                          mu=58.6539,
+                          sigma=5.5810,
+                          numspikes=1,
+                          location='distal',
+                          weights_ampa=weights_ampa_d1,
+                          weights_nmda=weights_nmda_d1,
+                          synaptic_delays=synaptic_delays_d1)
+
+# Proximal 2
+weights_ampa_p2 = {'L2_basket': 0.01, 'L2_pyramidal': 0.01, 'L5_basket': 0.01,
+                   'L5_pyramidal': 0.01}
+weights_nmda_p2 = {'L2_basket': 0.01, 'L2_pyramidal': 0.01, 'L5_basket': 0.01,
+                   'L5_pyramidal': 0.01}
+net_init.add_evoked_drive('evprox2',
+                          mu=80,
+                          sigma=1,
+                          numspikes=1,
+                          location='proximal',
+                          weights_ampa=weights_ampa_p2,
+                          weights_nmda=weights_nmda_p2,
+                          synaptic_delays=synaptic_delays_p)
+
+with MPIBackend(n_procs=n_procs, mpi_cmd='mpiexec'):
+    init_dpl = simulate_dipole(net_init, tstop=tstop, n_trials=1)[0]
+init_dpl.scale(scale_factor)
+init_dpl.smooth(smooth_window_len)
 
 ###############################################################################
 # Now we start the optimization!
-
-from hnn_core.optimization import optimize_evoked
-
-with MPIBackend(n_procs=n_procs):
-    net_opt = optimize_evoked(net, tstop=tstop, n_trials=1,
-                              target_dpl=exp_dpl, initial_dpl=initial_dpl,
-                              scale_factor=scale_factor,
-                              smooth_window_len=smooth_window_len)
+#
+# First, we define a function that will tell the optimization routine how to
+# modify the network drive parameters. The function will take in the Network
+# object with no attached drives, and a dictionary of the paramters we wish to
+# optimize.
+
+
+def set_params(net, params):
+
+    # Proximal 1
+    weights_ampa_p1 = {'L2_basket': 0.2913, 'L2_pyramidal': 0.9337,
+                       'L5_basket': 0.1951, 'L5_pyramidal': 0.3602}
+    weights_nmda_p1 = {'L2_basket': 0.9240, 'L2_pyramidal': 0.0845,
+                       'L5_basket': 0.5849, 'L5_pyramidal': 0.65105}
+    synaptic_delays_p = {'L2_basket': 0.1, 'L2_pyramidal': 0.1,
+                         'L5_basket': 1., 'L5_pyramidal': 1.}
```

Comment:
why do you redefine these?

Reply:
The only other option would be to pass in the initial network object (`net_init`) to `set_params`, and inside `set_params` we can tell the user to remove the last drive/the drive they will be optimizing (`evprox2`). Is that a good way to go?

Reply:
![image](https://github.com/jonescompneurolab/hnn-core/assets/51917715/521fb9b7-2341-4cee-a984-79ff52d59c9d)
![image](https://github.com/jonescompneurolab/hnn-core/assets/51917715/b4769a4a-8869-43f5-bd0f-c596c35a61e6)


Optimizing only one drive gives a convergence plot that still looks pretty flat, just because most of the dipole is explained by 2 of the drives. 

Would it be a good idea to have the initial dipole be the dipole that results from loading in the `default.json` parameters and running optimization on all drives? Optimizing all drives actually takes the same amount of time as optimizing one drive (since what takes long is `simulate_dipole`). 

Alternatively, we could keep optimizing only one drive but set `max_iter` to a smaller value like 50.